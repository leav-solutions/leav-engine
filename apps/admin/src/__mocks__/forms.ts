// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
import {GET_FORM_forms_list} from '../_gqlTypes/GET_FORM';
import {GET_FORMS_LIST_forms_list} from '../_gqlTypes/GET_FORMS_LIST';
import {FormElementTypes} from '../_gqlTypes/globalTypes';

const myField = {
    id: '123456',
    order: 0,
    uiElementType: 'field_container',
    containerId: 'some_container',
    settings: [
        {key: 'attribute', value: 'test_attribute'},
        {key: 'required', value: false},
        {key: 'input', value: 'text'}
    ],
    type: FormElementTypes.layout
};

export const mockFormLight: GET_FORMS_LIST_forms_list = {
    id: 'test_form_light',
    system: false,
    label: {fr: 'Test Form'}
};

export const mockFormFull: GET_FORM_forms_list = {
    ...mockFormLight,
    id: 'test_form_full',
    system: false,
    dependencyAttributes: [],
    label: {fr: 'Test Form'},
    elements: [{dependencyValue: null, elements: [{...myField}]}]
};
