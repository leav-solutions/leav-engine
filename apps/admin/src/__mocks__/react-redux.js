// Copyright LEAV Solutions 2017 until 2023/11/05, Copyright Aristid from 2023/11/06
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
module.exports = {
    connect: (mapStateToProps, mapDispatchToProps) => ReactComponent => ({
        mapStateToProps,
        mapDispatchToProps,
        ReactComponent
    }),
    useDispatch: jest.fn(),
    useSelector: jest.fn(),
    Provider: ({children}) => children
};
