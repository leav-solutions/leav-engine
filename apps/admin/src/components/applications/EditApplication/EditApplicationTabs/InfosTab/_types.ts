// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
import {GET_APPLICATION_BY_ID_applications_list} from '_gqlTypes/GET_APPLICATION_BY_ID';

export type ApplicationInfosFormValues = Omit<
    GET_APPLICATION_BY_ID_applications_list,
    'permissions' | 'url' | 'settings'
>;
