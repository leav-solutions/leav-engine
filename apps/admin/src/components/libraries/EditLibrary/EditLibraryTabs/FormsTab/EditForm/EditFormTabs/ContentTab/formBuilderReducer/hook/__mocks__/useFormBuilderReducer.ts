// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
import React, {Dispatch} from 'react';
import {FormBuilderAction, IFormBuilderState} from '../../formBuilderReducer';
import {mockInitialState} from '../../_fixtures/fixtures';

const initialState: IFormBuilderState = {
    form: null,
    library: '',
    openSettings: false,
    activeDependency: null,
    elementInSettings: null,
    elements: {},
    activeElements: {}
};

export const FormBuilderReducerContext = React.createContext<{
    state: IFormBuilderState;
    dispatch: Dispatch<FormBuilderAction>;
}>({state: mockInitialState, dispatch: () => jest.fn()});

export const useFormBuilderReducer = () => ({
    state: mockInitialState,
    dispatch: jest.fn()
});
