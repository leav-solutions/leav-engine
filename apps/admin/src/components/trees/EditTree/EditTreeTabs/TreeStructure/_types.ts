// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
import {GET_TREES_trees_list_libraries} from '_gqlTypes/GET_TREES';

export const LIBRARY_DND_TYPE = 'library';
export const ROOT_ID = '__root__';
export const ALLOW_ALL_ID = '__all__';

export interface IDndLibraryItem {
    type: string;
    from: string;
    library: GET_TREES_trees_list_libraries;
}

export interface IDndDropResult {
    id: string;
}
