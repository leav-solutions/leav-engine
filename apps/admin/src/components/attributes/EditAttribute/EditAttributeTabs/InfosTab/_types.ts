// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
import {
    GET_ATTRIBUTE_BY_ID_attributes_list,
    GET_ATTRIBUTE_BY_ID_attributes_list_StandardAttribute_versions_conf
} from '_gqlTypes/GET_ATTRIBUTE_BY_ID';
import {Override} from '_types/Override';

export type AttributeInfosFormValuesVersionsConf = Override<
    GET_ATTRIBUTE_BY_ID_attributes_list_StandardAttribute_versions_conf,
    {profile: string}
>;

export type AttributeInfosFormValues = Override<
    GET_ATTRIBUTE_BY_ID_attributes_list,
    {
        linked_library?: string;
        linked_tree?: string;
        reverse_link?: string;
        versions_conf: AttributeInfosFormValuesVersionsConf;
        unique: boolean;
    }
>;
