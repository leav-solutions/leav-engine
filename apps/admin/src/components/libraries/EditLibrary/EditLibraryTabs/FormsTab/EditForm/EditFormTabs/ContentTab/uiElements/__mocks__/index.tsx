// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
import React from 'react';
import {FieldTypes, IUIElement, UIElementTypes} from '../../_types';

function FakeComp(): JSX.Element {
    return <div>Mock</div>;
}

const baseMock = {
    component: <FakeComp />,
    canDrop: jest.fn(),
    settings: []
};

export const layoutElements: {[type in UIElementTypes]: IUIElement} = {
    [UIElementTypes.FIELDS_CONTAINER]: {
        ...baseMock,
        type: UIElementTypes.FIELDS_CONTAINER
    },
    [UIElementTypes.DIVIDER]: {
        ...baseMock,
        type: UIElementTypes.DIVIDER
    },
    [UIElementTypes.TEXT_BLOCK]: {
        ...baseMock,
        type: UIElementTypes.TEXT_BLOCK
    },
    [UIElementTypes.TABS]: {
        ...baseMock,
        type: UIElementTypes.TABS
    }
};

export const formElements: {[type in FieldTypes]: IUIElement} = {
    [FieldTypes.TEXT_INPUT]: {
        ...baseMock,
        type: FieldTypes.TEXT_INPUT
    },
    [FieldTypes.DATE]: {
        ...baseMock,
        type: FieldTypes.DATE
    },
    [FieldTypes.CHECKBOX]: {
        ...baseMock,
        type: FieldTypes.CHECKBOX
    },
    [FieldTypes.ENCRYPTED]: {
        ...baseMock,
        type: FieldTypes.ENCRYPTED
    },
    [FieldTypes.DROPDOWN]: {
        ...baseMock,
        type: FieldTypes.DROPDOWN
    },
    [FieldTypes.LINK]: {
        ...baseMock,
        type: FieldTypes.LINK
    },
    [FieldTypes.TREE]: {
        ...baseMock,
        type: FieldTypes.TREE
    }
};
