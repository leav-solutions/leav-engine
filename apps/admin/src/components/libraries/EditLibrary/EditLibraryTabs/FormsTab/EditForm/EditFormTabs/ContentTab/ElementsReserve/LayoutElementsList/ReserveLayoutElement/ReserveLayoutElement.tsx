// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
import React from 'react';
import {useDrag} from 'react-dnd';
import {useTranslation} from 'react-i18next';
import styled from 'styled-components';
import {v4 as uuidv4} from 'uuid';
import {FormElementTypes} from '../../../../../../../../../../../_gqlTypes/globalTypes';
import {useEditFormContext} from '../../../../../hooks/useEditFormContext';
import {defaultContainerId, FormBuilderActionTypes} from '../../../formBuilderReducer/formBuilderReducer';
import {useFormBuilderReducer} from '../../../formBuilderReducer/hook/useFormBuilderReducer';
import {DraggableElementTypes, IFormBuilderDragObject, IFormElement, IUIElement} from '../../../_types';

interface IReserveLayoutElementProps {
    element: IUIElement;
}

const Wrapper = styled.div<{isDragging: boolean}>`
    opacity: ${props => (props.isDragging ? 0.5 : 1)};
    font-weight: bold;
    cursor: move;
    margin: 1em 0;
`;

function ReserveLayoutElement({element}: IReserveLayoutElementProps): JSX.Element {
    const {dispatch} = useFormBuilderReducer();
    const {t} = useTranslation();
    const {readonly} = useEditFormContext();

    const formElement = {
        id: uuidv4(),
        order: 0,
        containerId: defaultContainerId,
        type: FormElementTypes.layout,
        uiElement: element
    };

    const [{isDragging}, drag] = useDrag<
        IFormBuilderDragObject<IFormElement>,
        IFormBuilderDragObject<IFormElement>,
        {isDragging: boolean}
    >({
        item: {
            type: DraggableElementTypes.RESERVE_LAYOUT_ELEMENT,
            element: formElement,
            index: -1
        },
        collect: monitor => ({
            isDragging: !!monitor.isDragging()
        }),
        canDrag: !readonly,
        end: (_, monitor) => {
            if (monitor.didDrop()) {
                // Item has already been added, don't do anything
                if (typeof monitor.getItem().dropAtPos !== 'undefined') {
                    return;
                }

                const position = {
                    order: monitor.getItem().dropAtPos?.order || 0,
                    containerId: monitor.getDropResult().containerId
                };

                dispatch({
                    type: FormBuilderActionTypes.ADD_ELEMENT,
                    element: {...monitor.getItem().element, containerId: position.containerId},
                    position
                });
            } else {
                const containerId = monitor.getItem().dropAtPos?.containerId || defaultContainerId;

                dispatch({
                    type: FormBuilderActionTypes.REMOVE_ELEMENT,
                    element: {...monitor.getItem().element, containerId}
                });
            }
        }
    });

    return (
        <Wrapper isDragging={isDragging} ref={drag}>
            {t(`forms.elements.${element.type}`)}
        </Wrapper>
    );
}

export default ReserveLayoutElement;
