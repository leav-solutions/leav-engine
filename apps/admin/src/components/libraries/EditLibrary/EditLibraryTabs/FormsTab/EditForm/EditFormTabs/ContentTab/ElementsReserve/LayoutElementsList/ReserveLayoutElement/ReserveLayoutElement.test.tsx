// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
import {render} from 'enzyme';
import React from 'react';
import {DndProvider} from 'react-dnd';
import {TestBackend} from 'react-dnd-test-backend';
import {layoutElements} from '../../../uiElements/__mocks__';
import {UIElementTypes} from '../../../_types';
import ReserveLayoutElement from './ReserveLayoutElement';

jest.mock('react', () => ({
    ...jest.requireActual('react'),
    useLayoutEffect: jest.requireActual('react').useEffect
}));

jest.mock('../../../formBuilderReducer/hook/useFormBuilderReducer');

describe('ReserveLayoutElement', () => {
    test('Snapshot test', async () => {
        const comp = render(
            <DndProvider backend={TestBackend}>
                <ReserveLayoutElement element={layoutElements[UIElementTypes.DIVIDER]} />
            </DndProvider>
        );

        expect(comp).toMatchSnapshot();
    });
});
