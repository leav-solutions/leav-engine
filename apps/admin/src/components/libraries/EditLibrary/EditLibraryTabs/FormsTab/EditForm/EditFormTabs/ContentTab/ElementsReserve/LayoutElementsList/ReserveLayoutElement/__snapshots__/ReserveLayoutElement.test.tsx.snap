// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReserveLayoutElement Snapshot test 1`] = `
LoadedCheerio {
  "0": Node {
    "attribs": {
      "class": "sc-bczRLJ iuQIus",
    },
    "children": [
      Node {
        "data": "forms.elements.divider",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": {
      "class": undefined,
    },
    "x-attribsPrefix": {
      "class": undefined,
    },
  },
  "_root": LoadedCheerio {
    "0": Node {
      "children": [
        Node {
          "attribs": {},
          "children": [
            Node {
              "attribs": {},
              "children": [],
              "name": "head",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": {},
                "children": [],
                "name": "body",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": {},
                "x-attribsPrefix": {},
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": {},
              "x-attribsPrefix": {},
            },
            Node {
              "attribs": {},
              "children": [],
              "name": "body",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Node {
                "attribs": {},
                "children": [],
                "name": "head",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": {},
                "x-attribsPrefix": {},
              },
              "type": "tag",
              "x-attribsNamespace": {},
              "x-attribsPrefix": {},
            },
          ],
          "name": "html",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": {},
          "x-attribsPrefix": {},
        },
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
